src/AoCElfTools.jl

```diff
@@ -1,5 +1,122 @@
 module AoCElfTools
 
-include("aoc2022.jl")
+# Project dependencies (from Base)
+using Printf
+
+# Abstract function to be extended for each day
+function parseday end
+function solveday end
+
+# Include utility functions
+include("utils.jl")
+
+export samplepath
+export userpath
+
+# Include solution code
+foreach(include, readdir(joinpath(@__DIR__, "2022"); join=true))
+
+"""
+    parse_input(i::Int) -> (IO -> puzzle input)
+
+Returns a function that parses Day `i`'s puzzle input into a data structure that can then
+be used to find a solution.
+
+Designed to be passed as a first argument to `open`.
+
+# Examples
+```jldoctest
+julia> input = "test/data/sample/01.txt";
+
+julia> println(read(input, String))
+1000
+2000
+3000
+
+4000
+
+5000
+6000
+
+7000
+8000
+9000
+
+10000
+
+julia> open(parse_input(1), input)
+5-element Vector{Int64}:
+  6000
+  4000
+ 11000
+ 24000
+ 10000
+```
+"""
+parse_input(i::Int) = parseday(Val(i))
+
+
+"""
+    solve(i::Int) -> (puzzle input -> solution)
+
+Returns a function that receive's Day `i`'s puzzle input and returns a solution as a `Tuple`
+containing the two answers.
+
+# Examples
+```jldoctest
```

Comment:
Doctests are failing here and in several updated places. I think I had to approve contributor PR triggers of CI runs on GitHub so these failures should now surface, however it would be good to check them locally before pushing.

Reply:
I've removed all of the broken doctests and successfully built the documentation locally on the last commit.